client/src/store.js

```diff
@@ -1,17 +1,27 @@
 import thunk from "redux-thunk";
 import { createStore, combineReducers, applyMiddleware } from "redux";
-import { userLoginReducer, userRegisterReducer } from "./state/reducers/userReducers";
+import {
+  userLoginReducer,
+  userRegisterReducer,
+} from "./state/reducers/userReducers";
+import { cartReducer } from "./state/reducers/cartReducer";
 
 const reducer = combineReducers({
   userLogin: userLoginReducer,
-  userRegister: userRegisterReducer
+  userRegister: userRegisterReducer,
+  cart: cartReducer,
 });
 
+const cartItemsFromStorage = localStorage.getItem("cartItems")
+  ? JSON.parse(localStorage.getItem("cartItems"))
+  : [];
```

Comment:
I don't think you need to save this data to localStorage you can just get it from the backend whenever a user log in